crates/sui-indexer/src/store/pg_indexer_store.rs

```diff
@@ -1248,6 +1258,39 @@ impl IndexerStore for PgIndexerStore {
         })
     }
 
+    async fn persist_object_balances(
+        &self,
+        checkpoint: i64,
+        object_balances: &[ObjectBalance],
+    ) -> Result<(), IndexerError> {
+        transactional_blocking!(&self.blocking_cp, |conn| {
+            // Commit indexed transactions
+            for object_balance_chunk in object_balances.chunks(PG_COMMIT_CHUNK_SIZE) {
+                diesel::insert_into(object_balances::table)
+                    .values(object_balance_chunk)
+                    .on_conflict_do_nothing()
+                    .execute(conn)
+                    .map_err(IndexerError::from)
+                    .context("Failed writing object balances to PostgresDB")?;
+            }
+
+            let watermark = Watermark {
+                name: crate::processors::object_balance_processor::OBJECT_BALANCES_WATERMARK
+                    .to_owned(),
+                checkpoint: Some(checkpoint),
+                epoch: None,
+            };
+
+            diesel::insert_into(watermarks::table)
+                .values(&watermark)
```

Comment:
iiuc, we want to keep only one row for `OBJECT_BALANCES_WATERMARK`, thus here we want to update the row?
sth like
```
                    .on_conflict(...)
                    .do_update()
                    .set(...)
```